outputs.tf

```diff
@@ -0,0 +1,9 @@
+output "id" {
+  value       = var.self_hosted_linux_agent_enable ? azuredevops_agent_pool.this[0].id : ""
```

Comment:
you can use 'try' here, for example - try(azuredevops_agent_pool.this[0].id, null)